app/lib/frontend/templates/layout.dart

```diff
@@ -103,40 +101,53 @@ String renderLayoutPage(
 
 String _renderSearchBanner({
   @required PageType type,
-  @required String platform,
-  @required String sdk,
   @required String publisherId,
   @required SearchQuery searchQuery,
 }) {
+  final sp = searchQuery != null
+      ? SearchPreference.fromSearchQuery(searchQuery)
+      : searchPreference ?? SearchPreference();
```

Comment:
Please put parens:
```
final sp = searchQuery != null
      ? SearchPreference.fromSearchQuery(searchQuery)
      : (searchPreference ?? SearchPreference());
```
Or however it is meant to be.
Otherwise it is hard to read the meaning.